aten/src/ATen/native/mkldnn/LinearAlgebra.cpp

```diff
@@ -0,0 +1,153 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+
+#if !AT_MKLDNN_ENABLED()
+namespace at {
+namespace native {
+
+at::Tensor mkldnn_mm(const at::Tensor& self, const at::Tensor& mat2) {
+  TORCH_CHECK(false, "mkldnn_mm: ATen not compiled with MKLDNN support");
+}
+
+at::Tensor& mkldnn_mm_out(const at::Tensor& self, const at::Tensor& mat2, at::Tensor& result) {
+  TORCH_CHECK(false, "mkldnn_mm_out: ATen not compiled with MKLDNN support");
+}
+
+at::Tensor& mkldnn_bmm_out(const at::Tensor& mat1, const at::Tensor& mat2, at::Tensor& result) {
+  TORCH_CHECK(false, "mkldnn_bmm_out: ATen not compiled with MKLDNN support");
+}
+
+Tensor& mkldnn_addmm_out(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha, Tensor &result) {
+  TORCH_CHECK(false, "mkldnn_addmm_out: ATen not compiled with MKLDNN support");
+}
+
+Tensor& mkldnn_addbmm_out(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha, Tensor &result) {
+  TORCH_CHECK(false, "mkldnn_addbmm_out: ATen not compiled with MKLDNN support");
+}
+
+Tensor& mkldnn_baddbmm_(Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha) {
+  TORCH_CHECK(false, "mkldnn_baddbmm_: ATen not compiled with MKLDNN support");
+}
+} // namespace native
+} // namespace at
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#include <ATen/native/mkldnn/Utils.h>
+#include <ATen/native/Resize.h>
+namespace at {
+namespace native {
+
+void matmul_common(
+    const Tensor &mat1,
+    const Tensor &mat2,
+    const Tensor &bias,
+    Tensor &result,
+    Scalar beta=1,
+    Scalar alpha=1,
+    const ideep::attr_t& attr = ideep::attr_t()) {
+  TORCH_CHECK(mat1.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK(mat2.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK(result.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+  TORCH_CHECK(mkldnn_bf16_device_check(),
+    "mkldnn_matmul: mkldnn_matmul bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
+
+  float dst_coeff = alpha.to<float>();
+  float sum_coeff = beta.to<float>();
+
+  // In OneDnn, 0 * nan = nan
+  // In aten, expect beta * result = 0 if beta = 0
+  if (sum_coeff == 0.0f)
+    result.zero_();
+  auto mat1_c = mat1.contiguous();
+  auto mat2_c = mat2.contiguous();
+  const ideep::tensor x = itensor_from_tensor(mat1_c);
+  const ideep::tensor w = itensor_from_tensor(mat2_c);
+  ideep::tensor y = itensor_from_tensor(result);
```

Comment:
it is possible that result is given in `addmm.out` as a non-contiguous buffer.
can we handle this correctly? aka. we can't change the output to a different buffer.